os/arch/arm/src/amebasmart/amebasmart_serial.c

```diff
@@ -1343,8 +1351,6 @@ int up_lowgetc(void)
  ****************************************************************************/
 int up_putc(int ch)
 {
-	/*check if there is space in fifo*/
-	while(!LOGUART_Ready());
 	/* Check for LF */
 
 	if (ch == '\n') {
```

Comment:
```
void LOGUART_PutChar(u8 c)
{
	LOGUART_TypeDef *UARTLOG = LOGUART_DEV;
	u32 CounterIndex = 0;
	u32 CPUID = SYS_CPUID();

	UARTLOG->LOGUART_UART_THRx[LOG_UART_IDX_FLAG[CPUID].idx] = c;

	if (c == KB_ASCII_LF) {
		UARTLOG->LOGUART_UART_THRx[LOG_UART_IDX_FLAG[CPUID].idx] = KB_ASCII_CR;
	}
}
```
What's the meaning of the if conditional? Does it append `\r`? or replace `\n` to `\r`?
If it's the append, then this line looks duplicated.

Reply:
Hi @sunghan-chang ,
It is append. This API will also be called from printf/dbg, if CR-LF is handled properly at all level log printing, I will remove this line.